pkg/daemon/ceph/osd/volume.go

```diff
@@ -57,15 +58,82 @@ func (a *OsdAgent) configureCVDevices(context *clusterd.Context, devices *Device
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate osd keyring. %+v", err)
 	}
-
-	if err = a.initializeDevices(context, devices); err != nil {
-		return nil, fmt.Errorf("failed to initialize devices. %+v", err)
+	if a.pvcBacked {
+		if lv, err = a.initializeBlockPVC(context, devices); err != nil {
+			return nil, fmt.Errorf("failed to initialize devices. %+v", err)
+		}
+	} else {
+		if err = a.initializeDevices(context, devices); err != nil {
+			return nil, fmt.Errorf("failed to initialize devices. %+v", err)
+		}
 	}
 
-	osds, err = getCephVolumeOSDs(context, a.cluster.Name, a.cluster.FSID)
+	osds, err = getCephVolumeOSDs(context, a.cluster.Name, a.cluster.FSID, lv)
 	return osds, err
 }
 
+func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *DeviceOsdMapping) (string, error) {
+	if err := sedChanges(context); err != nil {
+		return "", fmt.Errorf("sed failure, %+v", err) // fail return here as validation provided by ceph-volume
+	}
+	baseCommand := "stdbuf"
+	baseArgs := []string{"-oL", cephVolumeCmd, "lvm", "prepare"}
+	var lvpath string
+	for name, device := range devices.Entries {
+		if device.LegacyPartitionsFound {
+			logger.Infof("skipping device %s configured with legacy rook osd", name)
+			continue
+		}
+		if device.Data == -1 {
+			logger.Infof("configuring new device %s", name)
+			deviceArg := device.Config.Name
+			immediateExecuteArgs := append(baseArgs, []string{
+				"--data",
+				deviceArg,
+			}...)
+			// execute ceph-volume with the device
+
+			if op, err := context.Executor.ExecuteCommandWithOutput(false, "", baseCommand, immediateExecuteArgs...); err != nil {
+				return "", fmt.Errorf("failed ceph-volume. %+v", err) // fail return here as validation provided by ceph-volume
+			} else {
+				logger.Infof("%v", op)
+				vg := strings.Split(sys.Grep(op, "Volume group"), "\"")[1]
+				lv := strings.Split(sys.Grep(op, "Logical volume"), "\"")[1]
+				lvpath = fmt.Sprintf("/dev/%s/%s", vg, lv)
+			}
+		} else {
+			logger.Infof("skipping device %s with osd %d already configured", name, device.Data)
+		}
+	}
+
+	return lvpath, nil
+}
+
+func sedChanges(context *clusterd.Context) error {
```

Comment:
How about renaming this to `updateLVMConfig()`

Reply:
cc @sp98

Reply:
done